chat-widget/src/plugins/newMessageEventHandler.ts

```diff
@@ -1,112 +1,140 @@
+    const firstResponseLatencyTracker = new FirstResponseLatencyTracker();
+    // epoch time in utc for when start to listen.
+    // We dont longer have a mechanism to know if a message is history or new, so any message older than the time we start listening will be considered a history message.
+    // this is a workaround for the fact that we dont have a way to identify if a message is history or new, and it will provide consistency across different scenarios
+    const startTime = (new Date().getTime());
+    let isHistoryMessageReceivedEventRasied = false;
+
     const onNewAdapterActivityHandler = (activity: IActivity) => {
-        const isActivityMessage: boolean = activity?.type === Constants.message;
-        const isHistoryMessage: boolean = isActivityMessage && (activity?.channelData?.tags?.includes(Constants.historyMessageTag) || activity?.channelData?.fromList);
+        raiseMessageEvent(activity,);
+    };
+
+    const userSendMessageStrategy = (activity: IActivity) => {
+
+        const payload = buildMessagePayload(activity, userId);
+        // eslint-disable-next-line @typescript-eslint/no-explicit-any
+        payload.messageType = Constants.userMessageTag;
+        const newMessageSentEvent: ICustomEvent = {
+            eventName: BroadcastEvent.NewMessageSent,
+            payload: polyfillMessagePayloadForEvent(activity, payload, TelemetryManager.InternalTelemetryData?.conversationId)
+        };
+
+        BroadcastService.postMessage(newMessageSentEvent);
+
+        if (!isHistoryMessage(activity, startTime)) {
+            firstResponseLatencyTracker.startClock(payload);
+        }
 
-        raiseMessageEvent(activity, isHistoryMessage);
+        TelemetryHelper.logActionEvent(LogLevel.INFO, {
+            Event: TelemetryEvent.MessageSent,
+            Description: "New message sent"
+        });
     };
 
-    let isHistoryMessageReceivedEventRasied = false;
-    const raiseMessageEvent = (activity: IActivity, isHistoryMessage: boolean) => {
+    const systemMessageStrategy = (activity: IActivity) => {
+        const payload = buildMessagePayload(activity, userId);
+        // eslint-disable-next-line @typescript-eslint/no-explicit-any
+        payload.messageType = Constants.systemMessageTag;
+        TelemetryHelper.logActionEvent(LogLevel.INFO, {
+            Event: TelemetryEvent.SystemMessageReceived,
+            Description: "System message received"
+        });
+    };
+
+    const historyMessageStrategy = (payload: MessagePayload) => {
+
+        const newMessageReceivedEvent: ICustomEvent = {
+            eventName: BroadcastEvent.HistoryMessageReceived,
+            payload: payload
+        };
+        BroadcastService.postMessage(newMessageReceivedEvent);
+
+        if (!isHistoryMessageReceivedEventRasied) {
+            // this is needed for reload scenarios, it helps to identify the last message received before the reload
+            isHistoryMessageReceivedEventRasied = true;
```

Comment:
typo: `isHistoryMessageReceivedEventRasied` -> `isHistoryMessageReceivedEventRaised`

Reply:
in my defense, it was cut&paste :P